test/functional/helpers/spectrum.js

```diff
@@ -39,6 +40,11 @@ const createExpectError = selector => async t => {
     .ok("Expected field to have error when it did not");
 };
 
+const createExpectErrorByAttribute = selector => async t => {
+  await switchToIframe(t);
+  await t.expect(selector.getAttribute(invalidAttribute)).eql("true");
+};
```

Comment:
I understand why you did this. Ideally, we shouldn't have to have a way for checking for errors on a checkbox that's different than checking for errors on other components. I have some ideas for how to resolve this but I think it would require some changes (that should probably happen anyway) from react-spectrum. I've logged this issue to address it separately: https://jira.corp.adobe.com/browse/CORE-38529